integ-test/src/test/java/org/opensearch/sql/legacy/CsvFormatResponseIT.java

```diff
@@ -706,6 +706,103 @@ public void selectFunctionAsFieldTest() throws IOException {
     Assert.assertEquals(1, headers.size());
   }
 
+  @Test
+  public void unionTest() throws IOException {
+    String query =
+        String.format(
+            Locale.ROOT,
+            "SELECT firstname, lastname FROM %s LIMIT 3 "
```

Comment:
does UNION works for query with aggregation?

Reply:
Actually, UNION with Agg in legacy engine has bugs by my testing whatever format set. This fixing doesn't cover all existed bugs. You can try below queries.
```
POST _plugins/_sql?format=csv
{
  "query": "select customer_gender,count(type) from opensearch_dashboards_sample_data_ecommerce group by customer_gender UNION select customer_gender,count(type) from opensearch_dashboards_sample_data_ecommerce group by customer_gender"
}
POST _plugins/_sql?format=csv
{
  "query": "select count(type) from opensearch_dashboards_sample_data_ecommerce group by customer_gender UNION select count(type) from opensearch_dashboards_sample_data_ecommerce group by customer_gender"
}
```
May be we could open another ticket to fix the problems in Union with Agg? Or migrating to v2?

Reply:
And seems current implementation doesn't distinguish UNION and UNION ALL. A better way is to re-implement UNION in v2, PPL also requests multi-search implementation.
Again, this fixing is just for address the NPE problem of UNION query in JDBC format (such a common use case).